src/server/helpers/middleware.ts

```diff
@@ -72,6 +74,37 @@ export const loadLanguages = makeLoader('Language', 'languages', (a, b) => {
 	return a.name.localeCompare(b.name);
 });
 
+export function loadEntities(req: $Request, res: $Response, next: NextFunction) {
+	try {
+		const {entity} = res.locals;
+		if (entity.dataId) {
+			const {relationships} = entity;
+			relationships.forEach((relationship) => {
+				relationship.attributeSet.relationshipAttributes.forEach(attribute => {
+					relationship[`${attribute.type.name}`] = attribute.value.textValue || -Infinity;
+				});
+			});
```

Comment:
Setting the relationship attributes looks like it could maybe be done in `constructRelationships` ?

Reply:
I don't think we can use **constructRelationships**.
Here we are extracting the deeply nested number and position attribute value in the relationship object and then attaching directly to the relationship object. so that we can easily access it (e.g: relationship.number or relationship.position)

**Screenshot: Before Vs After**

![Screenshot from 2021-06-23 20-00-04](https://user-images.githubusercontent.com/55311336/123115375-b0e86780-d45d-11eb-9a8a-31ba5cdbd272.png)

Reply:
That was the wrong name, sorry. I was thinking of `loadEntityRelationships` defined in the same file.
The reason I'm suggesting to move the code to a more common method is that we are expecting other relationship attributes such as dates that won't have anything to do with Series and hence won't use the loadSeriesItems middleware.
Since you've already coded it to be generic (```relationship[`${attribute.type.name}`]```) it won't hurt to use it in loadEntityRelationships. I'd only suggest to wrap it in a condition just in case (`if(relationship.attributeSet?.relationshipAttributes)`)